### ISSUE TITLE
Support for <ol> to have line numbers

### ISSUE THREAD

**Issue by iafan** [sourcegraph/syntaxhighlight#20]
Would you be interested in accepting https://github.com/iafan/syntaxhighlight/commit/0da6819d25a1bdbf3c6206cc9ffece21d771bb57 as a PR?

This commit introduces `AsOrderedListHTML` function which, in addition to highlighting the source, wraps each line in `<li>..</li>`, and then adds `<ol>..</ol>` global wrapper. The resulting highlighted source code now has line numbers.

Commit comes with full test coverage.

**Comment by iafan** [comment]
Note that initially wanted to make as little changes to the code as possible and avoided changing the signature of `AsHTML()` function, but it might make more sense to use functional parameters here so that API could look like this:

```go
// default backward-compatible call
got, err := AsHTML(input)

// with optional parameter
got, err := AsHTML(input, OrderedList())
```

**Comment by sqs** [comment]
Absolutely. Go ahead and submit it. Thanks'

**Comment by iafan** [comment]
Great. Would you like me to adjust it to the proposed functional API above? Or do you prefer a separate `AsOrderedListHTML` function as in the current code?

**Comment by sqs** [comment]
The separate top-level function is fine with me. Whichever one you prefer.

**Comment by iafan** [comment]
Ok, I went ahead with functional params as this reduces the API surface and is more future-proof in case there will be other options.

**Comment by shurcooL** [comment]
I believe this issue was resolved in #21 and can be closed, is that right?

**Comment by sqs** [comment]
Yep, thanks!
